### ISSUE TITLE
[HELP] - Make NPM install node_modules in a child directory of project...

### ISSUE THREAD

**Issue by andrew-templeton** [npm/npm#13678]
#### I'm opening this issue because:

  - [X] npm is doing something I don't understand.

#### What's going wrong?

*npm v3.10.5*

I have a project which has a package.json / node_modules in the root of the project.

I have a `/build` directory which will contain many standalone node packages (which will be deployed to AWS Lambda), which MUST have their own `node_modules` and `package.json`.

When I execute `npm install` using the JS interface for npm (not the cli) during a build phase, I am trying to install the `node_modules` required for each of the subprojects in the `/build` directory, NOT use the ones in the parent / have npm try to skip installation in the subdirectory because it's already in a parent.

#### TLDR

How do I get this effect?

/    <-- Has packages
/build/foosubproj  <-- Needs *duplicates* of the packages

```
# while in /build/foosubproj
$ npm install
# Even when /node_modules has a dep declared in /build/foosubproj/package.json ,
#   I want npm to install those modules in /build/foosubproj/node_modules
```

**Comment by andrew-templeton** [comment]
This is essentially what I am trying to accomplish -> http://stackoverflow.com/questions/23270157/node-doesnt-install-child-module-dependencies-when-version-is-same-as-parent-mo

**Comment by andrew-templeton** [comment]
I ended up downgrading to NPM 2. This is really annoying behavior that is breaking for a whole class of build tools, there should be an option to ignore parent directories if you're not going to support a switch for nesting (which I know has been shot down in other issues).

**Comment by othiym23** [comment]
> When I execute `npm install` using the JS interface for npm (not the cli) during a build phase, I am trying to install the `node_modules` required for each of the subprojects in the `/build` directory, NOT use the ones in the parent / have npm try to skip installation in the subdirectory because it's already in a parent.

For your information, the challenges to using the API to do this kind of thing are significant, and is one of the major reasons that use of npm's API programmatically is unsupported. You're much better off running npm, regardless of its version, as a child process of your build tooling. In particular, trying to do more than one run in a given npm process isn't a good idea, as the internal state of the CLI is such that the consequences are more or less undefined. It may work fine with `npm@2` until you change something seemingly unrelated, at which point it will start failing in seemingly inexplicable ways. Please consider switching to the use of child processes. Thanks for your time, and for resolving this yourself!

**Comment by othiym23** [comment]
(BTW, when run as a child process, `npm --prefix <path/to/build/directory>` will always honor a `package.json` in that directory and not look higher in the tree)

**Comment by andrew-templeton** [comment]
Thanks for the comments! Will look into the `--prefix` flag. 

I believe the JS-vs-child-process part was in the end irrelevant to the issue, but thank you for the heads up re: unexpected behaviors. Thus far, I seem to not have issues, but I'll keep an eye out.
